### ISSUE TITLE
Unable to store varbinary(max) values larger than 8000 bytes

### ISSUE THREAD

**Issue by bkline** [mkleehammer/pyodbc#375]
This is an odd problem in that it only occurs under Python 3.x, but not under 2.7. I see the behavior with both the deprecated SQL Server Native Client NN.0 drivers as well as the recommended ODBC NN drivers, when running under Python 3.6. I can't reproduce the problem running under Python 2.7, no matter how large the value is. The threshold for the failure is 8001 bytes. Values 8000 bytes or smaller don't have a problem with either version of Python.

```
from argparse import ArgumentParser
from pyodbc import connect, Binary

parser = ArgumentParser()
parser.add_argument("--server", "-s", required=True)
parser.add_argument("--database", "-d", required=True)
parser.add_argument("--user", "-u", required=True)
parser.add_argument("--password", "-p", required=True)
parser.add_argument("--num_bytes", "-n", type=int, default=8001)
opts = parser.parse_args()
parms = dict(
    Driver="{ODBC Driver 13 for SQL Server}",
    Server=opts.server,
    Database=opts.database,
    Uid=opts.user,
    Pwd=opts.password
)
parms["Driver"] = "{SQL Server Native Client 11.0}"
b = b'x' * opts.num_bytes
connection_string = ";".join(["{}={}".format(*p) for p in parms.items()])
connection = connect(connection_string)
cursor = connection.cursor()
cursor.execute("CREATE TABLE #t (b VARBINARY(MAX))")
cursor.execute("INSERT INTO #t VALUES(?)", (Binary(b),))
cursor.execute("SELECT b FROM #t")
b2 = cursor.fetchone().b
print("OK" if b == b2 else "MISMATCH!")
```

With Python 2.7, or with a value of 8000 or less for the --num_bytes option, the output is `OK`.

With Python 3.6 and the default value of 8001 for the --num_bytes option, the output is:

```
Traceback (most recent call last):
  File "varbinary-repro.py", line 24, in <module>
    cursor.execute("INSERT INTO #t VALUES(?)", (Binary(b),))
pyodbc.Error: ('HY104', '[HY104] [Microsoft][SQL Server Native Client 11.0]Invalid precision value (0) (SQLBindParameter)')
```

Commenting out the `parms["Driver"] = ...` line gets the same outcome, except for the driver string displayed in the exception output.

Removing the `Binary()` cast makes no difference in the outcome.

I'm running pyodbc 4.0.22 under Python 3.6.2 (v3.6.2:5fd33b5, Jul  8 2017, 04:57:36) [MSC v.1900 64 bit (AMD64)] on Microsoft Windows [Version 6.1.7601] against Microsoft SQL Server 2016 (SP1-GDR) (KB4057118) - 13.0.4210.6 (X64)   Dec 22 2017 15:00:27   Copyright (c) Microsoft Corporation  Developer Edition (64-bit) on Windows Server 2012 R2 Standard 6.3 <X64> (Build 9600: ) (Hypervisor).

**Comment by gordthompson** [comment]
Changes in pyodbc 4.0.22 caused errors that have prompted numerous issues here. Version 2.0.23 reverted those problematic changes, so try upgrading to 4.0.23.

**Comment by bkline** [comment]
Ah! I see that our Python 2.7 version had pyodbc 4.0.17, which accounts for my not being able to reproduce the failures under the older Python. And sure enough, 4.0.23 eliminates the failures on Python 3.6. Thanks very much!

**Comment by bkline** [comment]
Problem has been corrected in 4.0.23.
